e2e/test/deployment.ts

```diff
@@ -0,0 +1,202 @@
+                .stderr(Error.deploymentConflict(deploymentName))
+                .run(Command.deploymentAdd(appName, deploymentName))
+                .end(done);
+        });
+        
+        it("fails if app not found", (done: MochaDone) => {
+            var fakeAppName: string = "not_a_real_app";
+            
+            nixt()
+                .stderr(Error.appNotFound(fakeAppName))
+                .run(Command.deploymentAdd(fakeAppName, deploymentName))
+                .end(done);
+        });
+    });
+    
+    describe("deployment rm", () => {
+        var deploymentName: string;
+        beforeEach((done: MochaDone) => {
+            deploymentName = makeRandomString();
+            
+            nixt()
+                .stdout(Success.deploymentAdd(appName, deploymentName))
+                .run(Command.deploymentAdd(appName, deploymentName))
+                .end(done);
+        });
+        
+        it("succeeds with Y", (done: MochaDone) => {
+            nixt()
+                .stdout(Success.deploymentRm(appName, deploymentName))
+                .run(Command.deploymentRm(appName, deploymentName))
+                .on(Command.PROMPT_ARE_YOU_SURE).respond(Command.RESPONSE_ACCEPT)
+                .end(() => {
+                    nixt()
+                        .expect(Validate.Deployments.checkFor(deploymentName, Validate.doesNotExistInContainer))
+                        .run(Command.deploymentLs(appName))
+                        .end(done);
+                });
+        });
+        
+        it("fails with n", (done: MochaDone) => {
+            nixt()
+                .stdout("Deployment removal cancelled.")
+                .run(Command.deploymentRm(appName, deploymentName))
+                .on(Command.PROMPT_ARE_YOU_SURE).respond(Command.RESPONSE_REJECT)
+                .end(() => {
+                    nixt()
+                        .expect(Validate.Deployments.checkFor(deploymentName, Validate.existsInContainer))
+                        .run(Command.deploymentLs(appName))
+                        .end(done);
+                });
+        });
+        
+        it("fails if app not found", (done: MochaDone) => {
+            var fakeAppName: string = "not_a_real_app";
+            
+            nixt()
+                .stderr(Error.appNotFound(fakeAppName))
+                .run(Command.deploymentRm(fakeAppName, deploymentName))
+                .on(Command.PROMPT_ARE_YOU_SURE).respond(Command.RESPONSE_ACCEPT)
+                .end(done);
+        });
+        
+        it("fails if deployment not found", (done: MochaDone) => {
+            var fakeDeploymentName: string = "not_a_real_deployment";
+            
+            nixt()
+                .stderr(Error.deploymentNotFound(fakeDeploymentName))
+                .run(Command.deploymentRm(appName, fakeDeploymentName))
+                .on(Command.PROMPT_ARE_YOU_SURE).respond(Command.RESPONSE_ACCEPT)
+                .end(done);
+        });
```

Comment:
nit: we should also add a test case for an `enter` key response, which is same as sending a `Y`